### PR TITLE
BUG: GH10747 where 'timestamp' is not inferred to be datetime column

### PR DESCRIPTION
**Pull request by kawochen**
addresses second point of #10747

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. can you add a small doc update in ``io.rst`` to specify what names are auto-inferred for this.

**Comment by kawochen** [comment]
Updated.

File: doc/source/io.rst [hunk 1]
```diff
@@ -1486,6 +1486,15 @@ be set to ``False`` if you need to preserve string-like numbers (e.g. '1', '2')
 
   Large integer values may be converted to dates if ``convert_dates=True`` and the data and / or column labels appear 'date-like'. The exact threshold depends on the ``date_unit`` specified.
 
+  'date-like' means that the column label meets at least one of the following case-insensitive criteria:
+
+     * it ends with ``'_at'``
+     * it ends with ``'_time'``
+     * it begins with ``'timestamp'``
+     * it is ``'modified'``
+     * it is ``'date'``
```
Comment (jreback) [inline doc/source/io.rst]: can u also add something like this to the doc-string

**Comment by kawochen** [comment]
updated

**Pull request merged**